### PR TITLE
Stripe

### PR DESCRIPTION
**Pull request by ajkatom**
### What this PR does (required):
- list PR functionality here

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/controllers/paymentMethod.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const stripe = require('../utils/stripe');
+const asyncHandler = require("express-async-handler");
+
+
+// @route POST /payment_methods
+// @desc create payment method
+// @access Public
+exports.createPaymentMethod = asyncHandler(async (req, res, next) => {
+  paymentMethod = stripe.createPaymentMethod(req.body);
```
Comment (moffatethan) [inline server/controllers/paymentMethod.js:9]: `const paymentMethod = . . .`

File: server/models/Payment.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  customerId: {
+    type: String,
+    required: true,
+  },
+});
+
+const getRate = async (sitterId) => {
+  const sitter = await Sitter.findById(sitterId);
+  return sitter.rate;
+};
```
Comment (moffatethan) [inline server/models/Payment.js:38]: I don't thing this is really needed, we could just grab the rate when fetching a pet sitter.

File: server/models/Payment.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+const payTotalOf = (rate, hoursOfService) => {
+  const processingFee = 5;
+  retrun(rate * hoursOfService + processingFee);
+};
```
Comment (moffatethan) [inline server/models/Payment.js:43]: Let's look at getters in Mongoose for this kind of calculation.

File: server/models/Payment.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+});
+
+const getRate = async (sitterId) => {
+  const sitter = await Sitter.findById(sitterId);
+  return sitter.rate;
+};
+
+const payTotalOf = (rate, hoursOfService) => {
+  const processingFee = 5;
+  retrun(rate * hoursOfService + processingFee);
```
Comment (moffatethan) [inline server/models/Payment.js:42]: `return`

**Review by moffatethan — CHANGES_REQUESTED** [review]
Good work just some quick comments 👍🏻

File: server/controllers/paymentMethod.js (hunk 1 above)
Comment (ajkatom) [reply server/controllers/paymentMethod.js:9]: you can see some of this was done at 1-3am lol

Comment (moffatethan) [reply server/controllers/paymentMethod.js:9]: Ahh 😂, no problem.

File: server/models/Payment.js (hunk 3 above)
Comment (ajkatom) [reply server/models/Payment.js:43]: fixed that in an old branch its good in the main branch
will address it here (i did all those fixes

File: server/models/Payment.js (hunk 4 above)
Comment (ajkatom) [reply server/models/Payment.js:42]: same ...fixed that in an old branch its good in the main branch
will address it here so there are no conflicts

File: server/models/Payment.js (hunk 2 above)
Comment (ajkatom) [reply server/models/Payment.js]: i changed that its a getter now

**Review by moffatethan — APPROVED** [review]

**Pull request merged**